### PR TITLE
Handle the case if site name is not given then will site name as “Glo…

### PR DESCRIPTION
**Pull request by Abhishek-121**
…bal” and fetch all devices under Global for getting device_uuid_list for swim distribute/activate operation.

In this commit -

— Handle the case if site name is not given then will site name as “Global” and fetch all devices under Global for getting device_uuid_list for swim distribute/activate operation.

— And then take the  intersection of device list API with given rules like - device role, device family and device series name.

### CODE REVIEW

File: plugins/modules/swim_intent.py [hunk 1]
```diff
@@ -741,8 +741,8 @@ def get_device_uuids(self, site_name, device_family, device_role, device_series_
 
         device_uuid_list = []
         if not site_name:
-            self.log("Failed to retrieve devices associated with the site due to missing site name", "INFO")
-            return device_uuid_list
+            site_name = "Global"
+            self.log("Since site name is not given so it will be fetch all the devices under Global and mark site name as 'Global'", "INFO")
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:745]: Site name not specified; defaulting to 'Global' to fetch all devices under this category

File: plugins/modules/swim_intent.py [hunk 2]
```diff
@@ -768,18 +767,56 @@ def get_device_uuids(self, site_name, device_family, device_role, device_series_
...
+        device_response = device_list_response.get('response')
+        if not response or not device_response:
+            self.log("Failed to retrieve devices associated with the site due to empty API response.")
+            return device_uuid_list
+
+        site_memberships_ids, device_response_ids = [], []
+
+        for item in site_response_list:
+            if item["reachabilityStatus"] != "Reachable":
+                self.log("""Reachability status of device '{0}' is '{1}', so cannot add it for distribution/activation
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:802]: the log messages for the both for loops are same.. How do we differentiate? Can we write below to make difference?

`self.log("Device '{0}' is currently '{1}' and cannot be included in the SWIM distribution/activation process.".format(item["managementIpAddress"], item["reachabilityStatus"]), "INFO")`

File: plugins/modules/swim_intent.py [hunk 3]
```diff
@@ -768,18 +767,56 @@ def get_device_uuids(self, site_name, device_family, device_role, device_series_
...
+                            task of swim image""".format(item["managementIpAddress"], item["reachabilityStatus"]), "INFO")
+                continue
+            self.log("""Successfully fetched the device '{0}' associated with the given site '{1}' for SWIM distribution/activation
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:805]: self.log("Device '{0}' from site '{1}' is ready for the SWIM distribution/activation process.".format(item["managementIpAddress"], site_name))

File: plugins/modules/swim_intent.py [hunk 4]
```diff
@@ -768,18 +767,56 @@ def get_device_uuids(self, site_name, device_family, device_role, device_series_
...
+                        task.""".format(item["managementIpAddress"], site_name))
+            site_memberships_ids.append(item["instanceUuid"])
+
+        for item in device_response:
+            if item["reachabilityStatus"] != "Reachable":
+                self.log("""Reachability status of device '{0}' is '{1}', so cannot add it for distribution/activation
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:811]: self.log("Unable to proceed with the device '{0}' for SWIM distribution/activation as its status is '{1}'.".format(item["managementIpAddress"], item["reachabilityStatus"]), "INFO")

File: plugins/modules/swim_intent.py [hunk 5]
```diff
@@ -768,18 +767,56 @@ def get_device_uuids(self, site_name, device_family, device_role, device_series_
...
+                            task of swim image""".format(item["managementIpAddress"], item["reachabilityStatus"]), "INFO")
+                continue
+            self.log("""Successfully fetched the device '{0}' with given device filter for SWIM distribution/activation
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:814]: `self.log("Device '{0}' matches to the specified filter requirements and is set for SWIM distribution/activation.".format(item["managementIpAddress"]))
`

File: plugins/modules/swim_intent.py [hunk 6]
```diff
@@ -768,18 +767,56 @@ def get_device_uuids(self, site_name, device_family, device_role, device_series_
...
+        device_list_response = self.dnac._exec(
+            family="devices",
+            function='get_device_list',
+            op_modifies=True,
+            params=device_params,
+        )
+
+        device_response = device_list_response.get('response')
+        if not response or not device_response:
+            self.log("Failed to retrieve devices associated with the site due to empty API response.")
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:795]: Can we print site name? What is the log level?

File: plugins/modules/swim_intent.py (hunk 1 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:745]: updated.

File: plugins/modules/swim_intent.py (hunk 2 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:802]: updated.

Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:802]: updated.

File: plugins/modules/swim_intent.py (hunk 3 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:805]: updated.

File: plugins/modules/swim_intent.py (hunk 4 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:811]: updated.

File: plugins/modules/swim_intent.py (hunk 5 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:814]: updated.

Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:814]: updated.

File: plugins/modules/swim_intent.py (hunk 6 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:795]: added log level.

**Review by madhansansel — APPROVED** [review]

**Pull request merged**